packages/frontend/src/features/metricsCatalog/components/MetricCatalogCategoryFormItem.tsx

```diff
@@ -124,19 +127,30 @@ export const MetricCatalogCategoryFormItem: FC<Props> = ({
     }
 
     return (
-        <Group spacing={4} position="apart" w="100%">
-            <CatalogCategory
-                category={category}
-                onClick={onClick}
-                onRemove={onDelete}
-            />
-            <ActionIcon
-                size="xs"
-                variant="subtle"
-                onClick={() => setIsEditing(true)}
-            >
-                <MantineIcon icon={IconDots} color="gray.6" size={14} />
-            </ActionIcon>
-        </Group>
+        <UnstyledButton
+            ref={hoverRef}
+            onClick={onClick}
+            px={4}
+            py={3}
+            sx={(theme) => ({
+                borderRadius: theme.radius.sm,
+                '&:hover': { backgroundColor: theme.colors.gray[2] },
+            })}
+        >
+            <Group spacing={4} position="apart" w="100%">
+                <CatalogCategory category={category} onRemove={onDelete} />
+                {hovered && (
+                    <ActionIcon
```

Comment:
```
validateDOMNesting(...): <button> cannot appear as a descendant of <button>. Error Component Stack
    at button (<anonymous>)

```

you could lift `ActionIcon` to be a sibling of `UnstyledButton` and with `pos` absolute it could be aligned at the end, wdyt?